### PR TITLE
Refactor GameChooserModel 'findByName' to return an optional instead of nullable

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: src/main/java/games/strategy/engine/framework/ui/GameChooserModel.java [hunk 1]
```diff
@@ -188,13 +188,16 @@ private static void addNewGameChooserEntry(final List<GameChooserEntry> entries,
...
+   */
+  public Optional<GameChooserEntry> findByName(final String name) {
     for (int i = 0; i < size(); i++) {
       if (get(i).getGameData().getGameName().equals(name)) {
-        return get(i);
+        return Optional.of(get(i));
       }
     }
-    return null;
+    return Optional.empty();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/engine/framework/ui/GameChooserModel.java]: Suggestion:
```java
return IntStream.range(0, size())
  .mapToObj(this::get)
  .filter(e -> e.getGameData.getGameName().equals(name))
  .findAny(); // Or findFirst if order does matter
```

Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/ui/GameChooserModel.java]: Good suggestion to use an int stream 👍   Done in: f65e0950a

File: src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java [hunk 2]
```diff
@@ -283,10 +283,10 @@ private GameChooserEntry selectByName(final boolean forceFactoryDefault) {
     final String userPreferredDefaultGameName = ClientSetting.DEFAULT_GAME_NAME_PREF.value();
 
     final GameChooserModel model = new GameChooserModel();
-    GameChooserEntry selectedGame = model.findByName(userPreferredDefaultGameName);
-    if (selectedGame == null) {
-      selectedGame = model.findByName(userPreferredDefaultGameName);
-    }
+    GameChooserEntry selectedGame = model.findByName(userPreferredDefaultGameName)
+        .orElse(model.findByName(userPreferredDefaultGameName)
+            .orElse(null));
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: I may be looking at this wrong, but it appears both the old and new code is calling `model.findByName(userPreferredDefaultGameName)` twice.  Can this just be simplified to the following:

```java
GameChooserEntry selectedGame = model.findByName(userPreferredDefaultGameName).orElse(null);
```

Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: It is the same thing twice! 0f72cdf14 removes the redundant call. I tried to track down the history, hard to tell if this was a simplification or an error.

**Pull request merged**